aeron-driver/src/main/c/media/aeron_udp_channel_transport_fixed_loss.c

```diff
@@ -0,0 +1,267 @@
+
+    return interceptor_bindings;
+}
+
+int aeron_udp_channel_interceptor_fixed_loss_configure(const aeron_udp_channel_interceptor_fixed_loss_params_t *fixed_loss_params)
+{
+    aeron_udp_channel_interceptor_fixed_loss_params = fixed_loss_params;
+
+    if (aeron_int64_counter_map_init(
+            &stream_and_session_id_to_offset_map, STREAM_AND_SESSION_ID_NULL_OFFSET, 16, AERON_MAP_DEFAULT_LOAD_FACTOR) < 0)
+    {
+        AERON_SET_ERR(errno, "%s", "Unable to init stream_and_session_id_to_offset_map");
+        return -1;
+    }
+
+    return 0;
+}
+
+void aeron_udp_channel_transport_fixed_loss_load_env(void)
+{
+    aeron_udp_channel_interceptor_fixed_loss_params_t *fixed_loss_params;
+    const char *args = AERON_CONFIG_GETENV_OR_DEFAULT(AERON_UDP_CHANNEL_TRANSPORT_BINDINGS_FIXED_LOSS_ARGS_ENV_VAR, "");
+    char *args_dup = strdup(args);
+    if (NULL == args_dup)
+    {
+        AERON_SET_ERR(errno, "%s", "Duplicating args string");
+        return;
+    }
+
+    if (aeron_alloc((void **)&fixed_loss_params, sizeof(aeron_udp_channel_interceptor_fixed_loss_params_t)) < 0)
+    {
+        return;
+    }
+
+    if (aeron_udp_channel_interceptor_fixed_loss_parse_params(args_dup, fixed_loss_params) >= 0)
+    {
+        // TODO handle return of -1
+        aeron_udp_channel_interceptor_fixed_loss_configure(fixed_loss_params);
+    }
+    else
+    {
+        aeron_free(fixed_loss_params);
+    }
+
+    aeron_free(args_dup);
+}
+
+int aeron_udp_channel_interceptor_fixed_loss_init_incoming(
+    void **interceptor_state, aeron_driver_context_t *context, aeron_udp_channel_transport_affinity_t affinity)
+{
+    (void)aeron_thread_once(&env_is_initialized, aeron_udp_channel_transport_fixed_loss_load_env);
+
+    if (NULL == aeron_udp_channel_interceptor_fixed_loss_params)
+    {
+        return -1;
+    }
+
+    *interceptor_state = NULL;
+
+    return 0;
+}
+
+static bool aeron_udp_channel_interceptor_fixed_loss_should_drop_frame(
+    const uint8_t *buffer, size_t buffer_length, const int32_t term_id, const int32_t term_offset, const size_t length)
+{
+    const aeron_frame_header_t *frame_header = (aeron_frame_header_t *)buffer;
+    const bool is_data_msg = (unsigned int)frame_header->type == AERON_HDR_TYPE_DATA;
+
+    if (is_data_msg)
+    {
+        const aeron_data_header_t *data_header = (aeron_data_header_t *)buffer;
+
+        if (term_id == data_header->term_id)
+        {
+            const int64_t key = aeron_map_compound_key(data_header->stream_id, data_header->session_id);
+
+            int64_t tracking_offset = aeron_int64_counter_map_get(&stream_and_session_id_to_offset_map, key);
+
+            if (tracking_offset == STREAM_AND_SESSION_ID_NULL_OFFSET)
+            {
+                tracking_offset = term_offset;
+                if (aeron_int64_counter_map_put(&stream_and_session_id_to_offset_map, key, tracking_offset, NULL) != 0)
+                {
+                    // TODO
```

Comment:
This is probably okay to just drop and return false.  The only practical failure here is ENOMEM, which we can't really recover from anyway.